src/lib/FlightTasks/tasks/FlightTaskAutoLine.cpp

```diff
@@ -0,0 +1,493 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file FlightAutoLine.cpp
+ */
+
+#include "FlightTaskAutoLine.hpp"
+#include <mathlib/mathlib.h>
+
+using namespace matrix;
+
+#define SIGMA_SINGLE_OP			0.000001f
+#define SIGMA_NORM			0.001f
+
+FlightTaskAutoLine::FlightTaskAutoLine(control::SuperBlock *parent, const char *name) :
+	FlightTaskAuto(parent, name),
+	_land_speed(parent, "MPC_LAND_SPEED", false),
+	_vel_max_up(parent, "MPC_Z_VEL_MAX_UP", false),
+	_vel_max_down(parent, "MPC_Z_VEL_MAX_DN", false),
+	_acc_max_xy(parent, "MPC_ACC_HOR_MAX", false),
+	_acc_xy(parent, "MPC_ACC_HOR", false),
+	_acc_max_up(parent, "MPC_ACC_UP_MAX", false),
+	_acc_max_down(parent, "MPC_ACC_DOWN_MAX", false),
+	_cruise_speed_90(parent, "MPC_CRUISE_90", false),
+	_nav_rad(parent, "NAV_ACC_RAD", false),
+	_mis_yaw_error(parent, "MIS_YAW_ERR", false)
+{}
+
+bool FlightTaskAutoLine::activate()
+{
+	_vel_sp_xy = matrix::Vector2f(&_velocity(0));
+	_pos_sp_xy = matrix::Vector2f(&_position(0));
+	_vel_sp_z = _velocity(2);
+	_pos_sp_z = _position(2);
+	_destination = _target;
+	_origin = _prev_wp;
+	_speed_at_target = 0.0f;
+
+	return FlightTaskAuto::activate();
+}
+
+bool FlightTaskAutoLine::update()
+{
+	if (_type == WaypointType::idle) {
+
+		_generateIdleSetpoints();
+
+	} else if (_type == WaypointType::land) {
+
+		_generateLandSetpoints();
+
+	} else if (_type == WaypointType::loiter || _type == WaypointType::position) {
+
+		_generateSetpoints();
+
+	} else if (_type == WaypointType::takeoff) {
+
+		_generateTakeoffSetpoints();
+
+	} else if (_type == WaypointType::velocity) {
+
+		_generateVelocitySetpoints();
+	}
+
+	/* Send setpoints */
+	_setVelocitySetpoint(Vector3f(_vel_sp_xy(0), _vel_sp_xy(1), _vel_sp_z));
+	_setPositionSetpoint(Vector3f(_pos_sp_xy(0), _pos_sp_xy(1), _pos_sp_z));
+	_setYawSetpoint(_yaw_wp);
+
+	return true;
+}
+
+void FlightTaskAutoLine::_generateIdleSetpoints()
+{
+	/* Send zero thrust setpoint */
+	_vel_sp_xy = matrix::Vector2f(NAN, NAN);
+	_vel_sp_z = NAN;
+	_pos_sp_xy = matrix::Vector2f(NAN, NAN);
+	_pos_sp_z = NAN;
+	_setThrustSetpoint(Vector3f(0.0f, 0.0f, 0.0f));
+}
+
+void FlightTaskAutoLine::_generateLandSetpoints()
+{
+	_pos_sp_xy = Vector2f(&_target(0));
+	_pos_sp_z = NAN;
+	_vel_sp_xy *= NAN;
```

Comment:
Multiply by NAN?

Reply:
that is just wrong. Luckily in this case it makes no difference since # * NAN = NAN.
By the way. I implemented the auto logic just recently. Therefore, before anyone is testing this PR, I need  to test if first myself on a real vehicle. 
By the way: if someone would test this PR the way it is implemented right now, he would just test the legacy code.